File: tests/fixtures/swift-deadcode-test/Sources/TrueDeadcode.swift
/// 這個文件包含真正的 deadcode（未使用的符號）
import Foundation

// MARK: - DEADCODE: 未使用的函式

/// 未使用的函式
func unusedFunction() -> String {
    return "This function is never called"
}

// MARK: - DEADCODE: 未使用的變數

/// 未使用的變數
let unusedVariable = "This variable is never referenced"

// MARK: - DEADCODE: 未使用的類別

/// 未使用的類別
class UnusedClass {
    private var data: String

    init(data: String) {
        self.data = data
    }

    func getData() -> String {
        return data
    }
}

// MARK: - DEADCODE: 未使用的常數

/// 未使用的常數
let UNUSED_CONSTANT = 42

// MARK: - DEADCODE: 未使用的結構體

/// 未使用的結構體
struct UnusedStruct {
    var value: Int

    func getValue() -> Int {
        return value
    }
}

// MARK: - DEADCODE: 未使用的列舉

/// 未使用的列舉
enum UnusedEnum {
    case optionA
    case optionB
    case optionC
}

// MARK: - DEADCODE: 未使用的協議

/// 未使用的協議
protocol UnusedProtocol {
    func doSomething()
}

// MARK: - DEADCODE: 未使用的內部類別

/// 未使用的內部類別
class UnusedInternalClass {
    static func helper() -> String {
        return "helper"
    }
}

// MARK: - 非 DEADCODE: 已使用的函式

/// 已使用的函式
func usedFunction() -> String {
    return "This function is used"
}

/// 內部使用的輔助函式
func internalHelper() -> String {
    return "helper"
}

/// 公開函式，使用 internalHelper
func publicFunction() -> String {
    return internalHelper()
}

// 使用 usedFunction 和 publicFunction
let _ = usedFunction()
let _ = publicFunction()

File: tests/fixtures/python-deadcode-test/true_deadcode.py
"""
這個文件包含真正的 deadcode（未使用的符號）
"""


# DEADCODE: 未使用的函式
def unused_function():
    """This function is never called"""
    return "This function is never called"


# DEADCODE: 未使用的變數
unused_variable = "This variable is never referenced"


# DEADCODE: 未使用的類別
class UnusedClass:
    """This class is never instantiated"""

    def __init__(self, data: str):
        self.data = data

    def get_data(self) -> str:
        return self.data


# DEADCODE: 未使用的常數
UNUSED_CONSTANT = 42


# DEADCODE: 未使用的 lambda
unused_lambda = lambda x: x * 2


# DEADCODE: 未使用的 async 函式
async def unused_async_function():
    """Unused async function"""
    return "async result"


# DEADCODE: 未使用的內部類別
class UnusedInternalClass:
    """Internal class that is never used"""

    @staticmethod
    def helper():
        return "helper"


# 非 DEADCODE: 已使用的函式
def used_function():
    """This function is used"""
    return "This function is used"


# 非 DEADCODE: 內部使用的函式
def internal_helper():
    """Helper function"""
    return "helper"


def public_function():
    """Public function that uses internal_helper"""
    return internal_helper()
